### PR TITLE
Add values to the bar plots for better visibility

### PR DESCRIPTION
**Pull request by leofang**
Right now we quote numbers from Fig.2, but some of the values are hard to read off, in particular those close to 1.0 because it's very close to the x-axis. By adding the values on top, the issue can be avoided.

### CODE REVIEW

File: paper.rst [hunk 1]
```diff
@@ -883,22 +883,22 @@ Python/NumPy `scipy.signal.welch()` function to use the array API.
...
 **making use of library specific performance optimizations**. The SciPy
 `welch()` implementation uses an optimization involving stride tricks. Stride
 tricks have not been standardized in the array API since they are not
-available in some libraries (e.g., JAX). NumPy, CuPy, and Torch allow setting
+available in some libraries (e.g., JAX). NumPy, CuPy, and PyTorch allow setting
 strides, but they do not use a uniform API to do so. An array API compatible
 implementation can be used, but it is slower, so it is used only as a fallback
 for libraries outside of NumPy, PyTorch, and CuPy. Indeed, it is significantly
-slower than even plain NumPy, with PyTorch CUDA taking 200 seconds to
+slower than even plain NumPy, with PyTorch GPU taking 200 seconds to
```
Comment (asmeurer) [inline paper.rst]: If this language is preferred it should be updated in the figure too.

**Comment by asmeurer** [comment]
The bar label in the third plot overlaps with the title.

**Comment by kgryte** [comment]
@leofang @asmeurer I'll add the values manually. The PDF and SVG currently in the paper are generated after post-processing in an SVG editor.

**Comment by asmeurer** [comment]
I would prefer to avoid that if possible. We'd have to redo it if I need to rerun the branchmarks (like for #15).

**Comment by kgryte** [comment]
Then we need to find a way to reliably add the `a-d` labels programmatically, so that the bar charts can be referred to individually.

File: paper.rst (hunk 1 above)
Comment (leofang) [reply paper.rst]: Ah, I thought the inconsistency I caught was only in the text, didn't pay attention to the plots 🤦🏻‍♂️ Yes, we should unify them.

**Comment by leofang** [comment]
If we move up the max from 80 to, say, 85 or 90, the label should fit in. Shall we do that in favor of (simple) automation?

**Comment by leofang** [comment]
> If we move up the max from 80 to, say, 85 or 90, the label should fit in. Shall we do that in favor of (simple) automation?

Done in [c714e23](https://github.com/data-apis/scipy-2023-presentation/pull/17/commits/c714e23a369bdf16446d1023b1210e1276412a45).

File: paper.rst (hunk 1 above)
Comment (leofang) [reply paper.rst]: I am looking at the latest main branch, and noticed that all references to "PyTorch CUDA" are gone. I'll update the figure.

Comment (leofang) [reply paper.rst]: Done in 6f20704.

File: benchmarks/generate_plot.py [hunk 2]
```diff
@@ -32,17 +32,21 @@
             data=scikit_learn_results[scikit_learn_results["Method"] ==
                                       "fit"], ax=ax1, errorbar=None)
 
+y_min = 0
+y_max = 82
```
Comment (kgryte) [inline benchmarks/generate_plot.py]: I think we can go ahead and make this 85, in order to give `78.5` a bit more breathing room.

Comment (leofang) [reply benchmarks/generate_plot.py]: Yeah that's what I shoot for originally, but tuned down a bit in case anyone prefers a more compact view 🙂

Comment (leofang) [reply benchmarks/generate_plot.py]: Updated in b7e835c.

**Review by kgryte — APPROVED** [review]

File: benchmarks/generate_plot.py (hunk 2 above)
Comment (kgryte) [reply benchmarks/generate_plot.py]: Thanks!

**Comment by leofang** [comment]
> Then we need to find a way to reliably add the `a-d` labels programmatically, so that the bar charts can be referred to individually.

I'm checking how this works
https://matplotlib.org/stable/gallery/text_labels_and_annotations/label_subplots.html
will send a follow-up PR once I figure out.

**Comment by kgryte** [comment]
Thanks, @leofang!

**Pull request merged**